packages/loader/container-loader/src/container.ts

```diff
@@ -816,26 +816,45 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             return;
         }
 
-        const previousContextState = await this.context.snapshotRuntimeState();
+        // By default, close the container and let the host reload.
+        // If hotSwapContext is true in the loader options, then try
+        // to reload the context without closing the container.
+        type ReloadState = { hotSwap: true; prevState: IRuntimeState } | { hotSwap: false };
+        let state: ReloadState = { hotSwap: false };
```

Comment:
>false [](start = 44, length = 5)

`this.options.hotSwapContext === true`
then you can remove the next line

Reply:
Moved it to the conditional.